livy_submit/livy_api.py

```diff
@@ -195,6 +197,19 @@ def submit(
                 continue
             data[var] = val
         #         print(data)
+
+        # check for collision with existing batch name
+        if name is not None:
+            _, _, batches = self.all_info()
+            for num,batch in batches.items():
+                if batch.name == name:
```

Comment:
is it possible that `batch.name` here could be `None`?

Reply:
Yes, I'll adjust to `if (batch.name is not None) and (batch.name == name)`